apollo-router/src/plugins/authentication.rs

```diff
@@ -353,37 +358,41 @@ impl Plugin for AuthenticationPlugin {
                         }
                     };
 
-                    request.context.enter_active_request().await;
-                    // Get the JWKS here
-                    let jwks_opt = match my_jwks.get(my_jwks_url).await {
-                        Ok(k) => k,
-                        Err(e) => {
-                            request.context.leave_active_request().await;
+                    // Search our set of JWKS to find the kid and process it
+                    // Note: This will search through JWKS in the order in which they are defined
+                    // in configuration. If the same "kid" is present in multiple JWKS, the router
+                    // will just use the first "kid" encountered.
+                    for jwks_url in my_jwks_urls {
+                        request.context.enter_active_request().await;
+                        // Get the JWKS here
+                        let jwks_opt = match my_jwks.get(jwks_url).await {
```

Comment:
This section of code looks like it will abort processing if one of the URLs does not work. Should it not proceed to try the other URLS?

Reply:
It's debatable. I think it should abort if it fails to retrieve one of the JWKS, since it could lead to confusing results (e.g.: "kid" is present in multiple JWKS and works until one day the JWKS can't be retrieved and then it tries to use a previously "shadowed" "kid" and starts to fail.).

I think it's one of the issues with multiple JWKS, although to be hones it's also a failing with a single JWKS, but that wouldn't have been a router problem with only single JWKS support.

Reply:
I don't know enough about the domain to comment on this, so I defer to your judgement here.